spec/unit/outputs/elasticsearch/http_client_spec.rb

```diff
@@ -287,4 +289,70 @@
       end
     end
   end
+
+  class StoppableServer
+
+    attr_reader :port
+
+    def initialize()
+      @port = rand(65535-1024) + 1025
+      queue = Queue.new
+      @first_req_waiter = java.util.concurrent.CountDownLatch.new(1)
+      @first_request = nil
+
+      @t = java.lang.Thread.new(
+        proc do
+          begin
+            @server = WEBrick::HTTPServer.new :Port => @port, :DocumentRoot => ".",
+                     :Logger => Cabin::Channel.get, # silence WEBrick logging
+                     :StartCallback => Proc.new {
+                           queue.push("started")
+                         }
+            @server.mount_proc '/headers_check' do |req, res|
+              res.body = 'Hello, world from WEBrick mocking server!'
+              @first_request = req
+              @first_req_waiter.countDown()
+            end
+
+            @server.start
+          rescue => e
+            puts "Error in webserver thread #{e}"
+            # ignore
+          end
+        end
+      )
+      @t.daemon = true
+      @t.start
+      queue.pop # blocks until the server is up
+    end
+
+    def stop
+      @server.shutdown
+    end
+
+    def wait_receive_request
+      @first_req_waiter.await(2, java.util.concurrent.TimeUnit::SECONDS)
+      @first_request
+    end
+  end
+
+  describe "#build_adapter" do
+    let(:client) { LogStash::Outputs::ElasticSearch::HttpClient.new(base_options) }
+    let!(:webserver) { StoppableServer.new } # webserver must be started before the call, so no lazy "let"
+
+    after :each do
+      webserver.stop
+    end
+
+    context "when don't customize 'user_agent' option then the 'user-agent' header" do
+      it "should contains the default Logstash and plugin version" do
+        adapter = client.build_adapter(client.options)
+        adapter.perform_request(::LogStash::Util::SafeURI.new("http://localhost:#{webserver.port}"), :get, "/headers_check")
+
+        request = webserver.wait_receive_request
+
+        expect(request.header['user-agent']).to include(client.prepare_user_agent)
```

Comment:
nit: we don't have any specs on `Client#prepare_user_agent`, so ensuring that the user agent we received includes the prepared user agent doesn't sufficiently safeguard us (for example, if `Client#prepare_user_agent` returned an empty string, _any_ string would match).

I would prefer that we set more refined expectations here.

```suggestion
        transmitted_user_agent = request.header['user-agent']
        
        expect(transmitted_user_agent).to include("Logstash/#{LOGSTASH_VERSION}")
        expect(transmitted_user_agent).to match(%r{logstash-output-elasticsearch/[0-9.]+}")
```